src/components/AbilityRotationVisualizer.tsx

```diff
@@ -17,23 +29,68 @@ const findNextIndex = (abilities: Ability[], currentIndex: number): number => {
   return nextIndex
 }
 
-export const AbilityRotationVisualizer = ({
-  size,
-}: AbilityRotationVisualizerProps) => {
-  const offSet = Math.floor(size / 2)
-  const [abilityRotation, setAbilityRotation] = useState<Ability[]>([])
+const getAbilitiesLeftSide = (
+  abilities: Ability[],
+  currentTick: number
+): number => {
+  const newArr = abilities.filter((el) => el.tick <= currentTick)
+  return newArr.length
+}
+
+export const AbilityRotationVisualizer = () => {
+  const [abilityRotation, setAbilityRotation] =
+    useState<Ability[]>(defaultRotation)
+
   const [magicState, setMagicState] = useState({
     currentIndex: 0,
     currentTick: 0,
+    useOffSet: false,
   })
   const [started, setStarted] = useState(false)
   const [abilityRotationName, setAbilityRotationName] = useState('')
+  const [size, setSize] = useState(0)
+  const [startPositions, setStartPositions] = useState(0)
+  const [blockContainerWidth, setBlockContainerWidth] = useState(0)
+
+  const elementRef = useRef<HTMLDivElement>(null)
+  const imgWidth = 50
+
+  useEffect(() => {
+    function handleResize() {
+      if (elementRef.current?.clientWidth) {
+        // Set how many abilities are showing (odd)
+        // 80 comes from 50px img width + 30px gap
+        setSize(Math.floor(elementRef.current.clientWidth / 80) | 1)
+        setStartPositions(elementRef.current.clientWidth / 2 - 25)
+        setBlockContainerWidth(
+          elementRef.current.clientWidth /
+            Math.floor(elementRef.current.clientWidth / 80)
+        )
+      }
+    }
+
+    if (elementRef.current?.clientWidth) {
+      setSize(Math.floor(elementRef.current.clientWidth / 80) | 1)
+      setStartPositions(elementRef.current.clientWidth / 2 - 25)
+      setBlockContainerWidth(
+        elementRef.current.clientWidth /
+          Math.floor(elementRef.current.clientWidth / 80)
+      )
+    }
```

Comment:
This block seems weird, you could just call `handleResize()`.

Reply:
True, done